File: app/src/main/java/com/youer/genshin/utils/CommonUtils.java
package com.youer.genshin.utils;

import java.text.DecimalFormat;

import com.youer.genshin.constants.AppendProp;

/**
 * @author youer
 * @date 9/26/23
 */
public class CommonUtils {

    /**
     * 展示圣遗物的属性
     *
     * @param key
     * @param value
     * @return
     */
    public static String displayRelicsValue(String key, double value) {
        if (AppendProp.getType(key).isPercent()) {
            return value + "%";
        }
        return format(value);
    }

    public static String format(double value){
        DecimalFormat decimalFormat = new DecimalFormat("#.#");
        return decimalFormat.format(value);
    }

    public static String convertUpCase(String camelCase) {
        StringBuilder result = new StringBuilder();

        for (char c : camelCase.toCharArray()) {
            if (Character.isUpperCase(c)) {
                result.append("_").append(Character.toUpperCase(c));
            } else {
                result.append(Character.toUpperCase(c));
            }
        }

        return result.toString();
    }

    public static String convertCamelCase(String uppercase) {
        StringBuilder result = new StringBuilder();
        boolean capitalizeNextChar = false;

        for (char c : uppercase.toCharArray()) {
            if (c == '_') {
                capitalizeNextChar = true;
            } else {
                if (capitalizeNextChar) {
                    result.append(Character.toUpperCase(c));
                    capitalizeNextChar = false;
                } else {
                    result.append(Character.toLowerCase(c));
                }
            }
        }

        return result.toString();
    }
} 

File: app/src/main/java/com/youer/genshin/utils/CommonUtils.kt
package com.youer.genshin.utils

import com.youer.genshin.constants.AppendProp
import java.text.DecimalFormat

/**
 * @author youer
 * @date 9/26/23
 */
object CommonUtils {
    /**
     * 展示圣遗物的属性
     *
     * @param key
     * @param value
     * @return
     */
    fun displayRelicsValue(key: String?, value: Double?): String {
        return if (AppendProp.getTypeByApiName(key)?.isPercent!!) {
            "$value%"
        } else format(value)
    }

    fun format(value: Double?): String {
        val decimalFormat = DecimalFormat("#.#")
        return decimalFormat.format(value)
    }

    fun convertUpCase(camelCase: String): String {
        val result = StringBuilder()
        for (c in camelCase.toCharArray()) {
            if (Character.isUpperCase(c)) {
                result.append("_").append(Character.toUpperCase(c))
            } else {
                result.append(Character.toUpperCase(c))
            }
        }
        return result.toString()
    }

    fun convertCamelCase(uppercase: String?): String {
        if (uppercase == null) {
            return ""
        }
        val result = StringBuilder()
        var capitalizeNextChar = false
        for (c in uppercase.toCharArray()) {
            if (c == '_') {
                capitalizeNextChar = true
            } else {
                if (capitalizeNextChar) {
                    result.append(Character.toUpperCase(c))
                    capitalizeNextChar = false
                } else {
                    result.append(Character.toLowerCase(c))
                }
            }
        }
        return result.toString()
    }
}